rc/filetype/css.kak

```diff
@@ -32,28 +33,78 @@ provide-module css %[
 
 # Highlighters
 # ‾‾‾‾‾‾‾‾‾‾‾‾
-
 add-highlighter shared/css regions
-add-highlighter shared/css/selector default-region group
-add-highlighter shared/css/declaration region [{] [}]  regions
-add-highlighter shared/css/comment    region /[*] [*]/ fill comment
+add-highlighter shared/css/code default-region group
+add-highlighter shared/css/attr_selector region \[ \] regions
+add-highlighter shared/css/double_string region '"'  (?<!\\)(\\\\)*" fill string
+add-highlighter shared/css/single_string region "'"  (?<!\\)(\\\\)*' fill string
+add-highlighter shared/css/comment    region /\* \*/ fill comment
+
+evaluate-commands %sh{
+  # html tags
+  # generated from the URL & <code> below
+  # includes elements that cannot be styled, which is fine.
+  # 
+  # https://developer.mozilla.org/en-US/docs/Web/HTML/Element
+  html_tags='html body address article aside footer header h1 h2 h3 h4 h5 h6 main nav section blockquote dd div dl dt figcaption figure hr li ol p pre ul a abbr b bdi bdo br cite code data dfn em i kbd mark q rp rt ruby s samp small span strong sub sup time u var wbr area audio img map track video embed iframe object param picture portal source canvas noscript script del ins caption col colgroup table tbody td tfoot th thead tr button datalist fieldset form input label legend meter optgroup option output progress select textarea details dialog menu summary slot template acronym applet basefont bgsound big blink center content dir font frame frameset hgroup image keygen marquee menuitem nobr noembed noframes plaintext rb rtc shadow spacer strike tt xmp'
+
+  # Units
+  # ‾‾‾‾‾
+  # generated from the URL & <code> below
+  # includes #rgb, #rrggbb, #rrggbbaa as color values {3,8}
+  # 
+  # https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units
+  units='% cap ch cm deg dpcm dpi dppx em ex grad Hz ic in kHz lh mm ms pc pt px Q rad rem rlh s turn vb vh vi vmax vmin vw x'
+
+  logical_ops='and not only from to'
+  keywords='!important auto inherit initial unset none'
+  media_types='all print screen speech'
+
+  # easing_re='linear|ease(-(in-out|in|out))?|step-(start|end)'
+
+  # combinators='+ > ~ ||'
+  # attribute_op='= ~= |= ^= $= *='
+
+  join() { eval set -- $1; IFS="|"; echo "$*"; }
+
+  # Selectors
+  # ‾‾‾‾‾‾‾‾‾
+  # universal: *, ns|*, *|*, |*
+  # class/id: .class, #id
+  # type: element
+  # attr: [attr=val]
+
+  # order below matters
+  printf %s "
+  add-highlighter shared/css/code/tag_selectors regex \b($(join "${html_tags}"))((:[a-z:])|[\h.#]) 1:keyword
```

Comment:
This change makes HTML tags get highlighted as "keyword" wherever they appear, if that helps. It looks OK for me with your `test.css`, but I assume there was a *reason* you tried to make HTML tags only recognised when followed by a psuedo-class, class, ID, or space.
```suggestion
  add-highlighter shared/css/code/tag_selectors regex \b$(join "${html_tags}")\b 0:keyword
```

Reply:
Yes, there is a reason. Simple html tags (like one letter ones) will cause problems, here on the screenshot I am unsure why it's not respecting word boundaries... but I did have it working and was causing problems when a simple tag (like a, b, i) would be preceded or followed by say a hyphen.

Here is what will happen if I follow your suggestion (copied your suggestion exactly as you have it). Actually I have added a comma recently, and just need to figure out how to have it be followed by a '[' which I somehow can't escape.
![image](https://user-images.githubusercontent.com/1596691/162959809-e30c4798-0f8e-4abf-b320-e7572b4ffa1d.png)

Reply:
This still needs the question, and help of determining how to have a tag be followed by '['... I just can't find how to escape it! Could you help me sort this out :)?

Reply:
> Simple html tags (like one letter ones) will cause problems, here on the screenshot I am unsure why it's not respecting word boundaries...

Ah, right, I see. `\b$(join "$html_tags")\b` will expand to something like:

    html|b|body

which turns the regex into:

    \bhtml|b|body\b

...which will match `html` at the beginning of a word, `body` at the end of a word, or `b` anywhere. I removed one too many grouping brackets, so you need to put one back:

    \b($(join "$html_tags"))\b

...which will expand to:

    \b(html|b|body)\b

...so that HTML tags are always required to be standalone words, not appearing within some other word.

Reply:
That still causes issues in cases like this, the word boundary I am afraid is not enough:
![image](https://user-images.githubusercontent.com/1596691/164111208-4383fdcd-8ef1-4698-bc1a-b55f774c353d.png)